### DISCUSSION TITLE
Unit Testing Query<T>

### DISCUSSION THREAD

**Discussion by pollbytes** [JasperFx/marten#3428] [General] [answered]
I am writing some xunit unit tests for marten.
I want to mock, using Moq package, the behavior of:

await session.Query<Account>().ToListAsync(token).ConfigureAwait(false);

but seems that it's not possible to cast an IQueryable object to type 'Marten.Linq.IMartenQueryable'.

Does anyone knows how to approach this?

Thanks

**Comment by mysticmind** [comment]
If you look at the tests we have Marten in test suite, everything pertaining to querying is fully integration tests with Postgres DB. @jeremydmiller always recommended *not to mock* the portions/LINQ support which you are attempting to do.

Our suggested approach is to write integration tests talking to a Postgres database.

**Comment by jeremydmiller** [comment, marked as answer]
@leopoldev404 I'll go even farther, I very, very strongly recommend against trying to mock out any kind of IQueryable implementation.
